wasm/jit/jit_value_location.go

```diff
@@ -0,0 +1,287 @@
+package jit
```

Comment:
This is just extracted the GOARCH independent parts of code from `jit_value_location_amd64.go` without any change!